components/console_cmd_ifconfig/console_ifconfig.c

```diff
@@ -0,0 +1,647 @@
+        } else {
+            ESP_LOGI(TAG, "Invalid argument: %s", argv[self->start_index + 1]);
+        }
+
+        return ESP_FAIL;
+    }
+#endif
+    ESP_LOGE(TAG, "NAPT not enabled in menuconfig");
+    return ESP_OK;
+}
+
+
+esp_err_t ifcfg_addr_op(netif_op *self, int argc, char *argv[], esp_netif_t *esp_netif)
+{
+    if (!strcmp("staticip", argv[self->start_index])) {
+        esp_netif_dhcpc_stop(esp_netif);
+        //esp_netif_dhcps_stop(esp_netif);
+        return ESP_OK;
+    } else if (!strcmp("server", argv[self->start_index + 1])) { // Server
+        if (!strcmp("enable", argv[self->start_index + 2])) {
+            ESP_LOGW(TAG, "DHCP Server configuration is not supported yet.");    // TBD
+            //esp_netif_dhcps_start(esp_netif);
+            return ESP_OK;
+        } else if (!strcmp("disable", argv[self->start_index + 2])) {
+            ESP_LOGW(TAG, "DHCP Server configuration is not supported yet.");    // TBD
+            //esp_netif_dhcps_stop(esp_netif);
+            return ESP_OK;
+        }
+
+        ESP_LOGE(TAG, "Invalid argument");
+        return ESP_FAIL;
+
+    } else if (!strcmp("client", argv[self->start_index + 1])) { // Client
+        if (!strcmp("enable", argv[self->start_index + 2])) {
+            esp_netif_dhcpc_start(esp_netif);
+            return ESP_OK;
+        } else if (!strcmp("disable", argv[self->start_index + 2])) {
+            esp_netif_dhcpc_stop(esp_netif);
+            return ESP_OK;
+        }
+
+        ESP_LOGE(TAG, "Invalid argument");
+        return ESP_FAIL;
+    }
+
+    return ESP_FAIL;
+}
+
+
+void print_iface_details(esp_netif_t *esp_netif)
+{
+    esp_netif_ip_info_t ip_info;
+    uint8_t mac[NETIF_MAX_HWADDR_LEN];
+    char interface[10];
+    int ip6_addrs_count = 0;
+    esp_ip6_addr_t ip6[LWIP_IPV6_NUM_ADDRESSES];
+    esp_err_t ret = ESP_FAIL;
+    esp_netif_dhcp_status_t status;
+
+    struct netif *lwip_netif = esp_netif_get_netif_impl(esp_netif);
+
+    /* Print Interface Name and Number */
+    ret = esp_netif_get_netif_impl_name(esp_netif, interface);
+    if ((ESP_FAIL == ret) || (NULL == esp_netif)) {
+        ESP_LOGE(TAG, "No interface available");
+        return;
+    }
+
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 1, 0)
+    if (esp_netif_get_default_netif() == esp_netif) {
+        ESP_LOGI(TAG, "Interface Name: %s (DEF)", interface);
+    } else {
+        ESP_LOGI(TAG, "Interface Name: %s", interface);
+    }
+#else
+    ESP_LOGI(TAG, "Interface Name: %s", interface);
+#endif
+    ESP_LOGI(TAG, "Interface Number: %d", lwip_netif->num);
+
+    /* Print MAC address */
+    esp_netif_get_mac(esp_netif, mac);
+    ESP_LOGI(TAG, "MAC: %02x:%02x:%02x:%02x:%02x:%02x", mac[0], mac[1],
+             mac[2], mac[3], mac[4], mac[5]);
+
+    /* Print DHCP status */
+    if (ESP_OK == esp_netif_dhcps_get_status(esp_netif, &status)) {
+        ESP_LOGI(TAG, "DHCP Server Status: %s", status ? "enabled" : "disabled");
```

Comment:
this `status` type is `esp_netif_dhcp_status_t`, which is an enum, you cannot simply convert it to `bool`